### PR TITLE
feat: add transient storage cleaning for InputVerifier

### PR DESCRIPTION
**Pull request by jatZama**

### CODE REVIEW

**Review by immortal-tofu — APPROVED** [review]
LGTM

**Comment by jatZama** [comment]
> Also, I think (I brought it up in the past) it would be good to have something such as `ACL is IACL.sol`, etc.

This will be part of another issue that I already opened about various optimizations we could add, I didn't plan to close this issue now. This PR is to close those 3 issues: #188 and #198 and #204 
What you are suggesting could be for example part of this issue (not supposed to be closed by this PR but later): #191

File: contracts/examples/PaymentLimit.sol [hunk 1]
```diff
@@ -4,16 +4,14 @@ pragma solidity ^0.8.24;
 
 import "../lib/TFHE.sol";
 import "@openzeppelin/contracts/access/Ownable2Step.sol";
```
[the remark this answers is not in the document]
Comment (jatZama) [reply contracts/examples/PaymentLimit.sol:6]: this is just an example contract for test but OK I will remove it now

File: contracts/lib/Impl.sol [hunk 2]
```diff
@@ -365,11 +365,16 @@ library Impl {
...
 
-    function cleanTransientStorage() internal {
+    function cleanTransientStorageACL() internal {
         FHEVMConfig.FHEVMConfigStruct storage $ = getFHEVMConfig();
         IACL($.ACLAddress).cleanTransientStorage();
     }
 
+    function cleanTransientStorageInputVerifier() internal {
+        FHEVMConfig.FHEVMConfigStruct storage $ = getFHEVMConfig();
+        IACL($.InputVerifierAddress).cleanTransientStorage();
```
[the remark this answers is not in the document]
Comment (jatZama) [reply contracts/lib/Impl.sol:375]: wdym by wrong interface? this function exists in IACL, so I don't understand.

**Comment by jatZama** [comment]
Closes #188 and #198 and #204

**Pull request merged**